deployment/ccip/changeset/v1_6/cs_add_new_chain_e2e.go

```diff
@@ -289,7 +297,20 @@ func addCandidatesForNewChainLogic(e deployment.Environment, c AddCandidatesForN
 	if err != nil {
 		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
 	}
-	donID, err := state.Chains[c.HomeChainSelector].CapabilityRegistry.GetNextDONId(&bind.CallOpts{
+
+	if c.DonIDOffSet != nil {
```

Comment:
a check for if the value is not zero

Reply:
donIDOffSet can be zero here, which syncs it's state to latest capReg